hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/test/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicationTaskTest.java

```diff
@@ -232,4 +241,73 @@ public void testCreateRowMutationsPutAndDeleteAlternate() throws IOException {
+
+    put1 = getPutCellCustomSize(10000, ROW_KEY, CF1, COL_QUALIFIER, TIMESTAMP);
+    put2 = getPutCellCustomSize(10000, ROW_KEY, CF1, COL_QUALIFIER_2, TIMESTAMP);
+    put3 = getPutCellCustomSize(10000, ROW_KEY, CF2, COL_QUALIFIER, TIMESTAMP);
+    RowMutations rowMutationsLargerSize =
+        CloudBigtableReplicationTask.buildRowMutations(ROW_KEY, Arrays.asList(put1, put2, put3));
+
+    boolean logAndSkipIncompatibleRowMutationsDoesNotExceed =
+        CloudBigtableReplicationTask.verifyRowMutationThresholds(
+            rowMutationsSmallerSize, conf, metricsExporter);
+    boolean logAndSkipIncompatibleRowMutationsExceeds =
+        CloudBigtableReplicationTask.verifyRowMutationThresholds(
+            rowMutationsLargerSize, conf, metricsExporter);
+    Assert.assertEquals(false, logAndSkipIncompatibleRowMutationsDoesNotExceed);
+    Assert.assertEquals(true, logAndSkipIncompatibleRowMutationsExceeds);
+  }
+
+  @Test
+  public void testFilterMaxCellsPerMutationExceedsThreshold() throws IOException {
+    Configuration conf = new Configuration();
+    conf.set(FILTER_MAX_CELLS_PER_MUTATION_KEY, "true");
+    conf.setInt(FILTER_MAX_CELLS_PER_MUTATION_THRESHOLD_KEY, 2);
+
+    Cell put1 = getPutCellCustomSize(10, ROW_KEY, CF1, COL_QUALIFIER, TIMESTAMP);
+    RowMutations rowMutationsCellsLessThanMax =
+        CloudBigtableReplicationTask.buildRowMutations(ROW_KEY, Arrays.asList(put1));
+
+    put1 = getPutCellCustomSize(10000, ROW_KEY, CF1, COL_QUALIFIER, TIMESTAMP);
+    Cell put2 = getPutCellCustomSize(10000, ROW_KEY, CF1, COL_QUALIFIER_2, TIMESTAMP);
+    RowMutations rowMutationsCellsEqualToMax =
+        CloudBigtableReplicationTask.buildRowMutations(ROW_KEY, Arrays.asList(put1, put2));
+
+    put1 = getPutCellCustomSize(10000, ROW_KEY, CF1, COL_QUALIFIER, TIMESTAMP);
+    put2 = getPutCellCustomSize(10000, ROW_KEY, CF1, COL_QUALIFIER_2, TIMESTAMP);
+    Cell put3 = getPutCellCustomSize(10, ROW_KEY, CF2, COL_QUALIFIER, TIMESTAMP);
+    RowMutations rowMutationsCellsGreaterThanMax =
+        CloudBigtableReplicationTask.buildRowMutations(ROW_KEY, Arrays.asList(put1, put2, put3));
+
+    boolean logAndSkipIncompatibleRowMutationsDoesNotExceed =
+        CloudBigtableReplicationTask.verifyRowMutationThresholds(
+            rowMutationsCellsLessThanMax, conf, metricsExporter);
+    boolean logAndSkipIncompatibleRowMutationsDoesNotExceedEqual =
+        CloudBigtableReplicationTask.verifyRowMutationThresholds(
+            rowMutationsCellsEqualToMax, conf, metricsExporter);
+    boolean logAndSkipIncompatibleRowMutationsExceeds =
+        CloudBigtableReplicationTask.verifyRowMutationThresholds(
+            rowMutationsCellsGreaterThanMax, conf, metricsExporter);
+    Assert.assertEquals(false, logAndSkipIncompatibleRowMutationsDoesNotExceed);
+    Assert.assertEquals(false, logAndSkipIncompatibleRowMutationsDoesNotExceedEqual);
+    Assert.assertEquals(true, logAndSkipIncompatibleRowMutationsExceeds);
+  }
+
```

Comment:
Add a test that validates that ReplicationTask does indeed use the results of the VerifyRowMutationThresholds. 

I think we should make VerifyRowMutationThresholds private and test that the task failed or succeeded and verify that the appropriate metrics were incremented.